packages/datetime/src/dateRangeInput.tsx

```diff
@@ -304,10 +342,16 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
 
     private handleStartInputChange = (e: React.FormEvent<HTMLInputElement>) => {
         this.handleInputChange(e, DateRangeBoundary.START);
+        if (this.props.startInputProps != null) {
+            Utils.safeInvoke(this.props.startInputProps.onChange, e);
+        }
     }
 
     private handleEndInputChange = (e: React.FormEvent<HTMLInputElement>) => {
         this.handleInputChange(e, DateRangeBoundary.END);
+        if (this.props.endInputProps != null) {
```

Comment:
I admittedly didn't test this case here, but I believe when you pass `null` to a prop, `defaultValues` isn't triggered. See: http://stackoverflow.com/questions/34809178/react-js-default-prop-is-not-used-with-null-is-passed. So this case defends against that (assuming it's true).